examples/vector_database/scripts/compute_vectors.py

```diff
@@ -0,0 +1,240 @@
+import abc
+import asyncio
+import base64
+from io import BytesIO
+import logging
+import os
+import pickle
+import shutil
+from typing import (Any, AsyncIterator, Dict, Generic, List, Optional, Tuple,
+                    TypeVar)
+
+import numpy as np
+import pandas as pd
+from PIL import Image
+import torch
+from tqdm import tqdm
+
+InputType = TypeVar('InputType')
+OutputType = TypeVar('OutputType')
+ModelInputType = TypeVar('ModelInputType')
+ModelOutputType = TypeVar('ModelOutputType')
+
+
+class BatchProcessor(Generic[InputType, OutputType], abc.ABC):
+    """Process ImageNet images with CLIP."""
```

Comment:
Include in the docstring a short version of the flow, e.g.:
1. setup_model initializes the model
2. get_dataset_iterator will yield individual items from the dataset
3. do_data_loading will get an item from the dataset iterator and do any preprocessing
4. the loaded items will be batched and handed to do_batch_processing for the ultimate processing